cmake/protobuf-configure-target.cmake

```diff
@@ -32,9 +32,7 @@ function(protobuf_configure_target target)
         # Allow big object
         target_compile_options("${target}" PRIVATE /bigobj)
     endif ()
-    if (protobuf_UNICODE)
-        target_compile_definitions("${target}" PRIVATE -DUNICODE -D_UNICODE)
-    endif ()
+    target_compile_definitions("${target}" PRIVATE -DUNICODE -D_UNICODE)
```

Comment:
This will affect all platforms, is this useful anywhere except MSVC?

Reply:
This two macros are used in Windows C runtime. See 
```
https://learn.microsoft.com/en-us/cpp/text/unicode-programming-summary?view=msvc-170
```

Reply:
makes sense, we should guard this on MSVC is we're not going to guard it on protobuf_UNICODE anymore then.  Otherwise this will be a newly added macro for linux and macos that could cause issues